lib/ansible/modules/database/postgresql/postgresql_vacuum.py

```diff
@@ -0,0 +1,371 @@
+    - Switch to session_role after connecting.
+      The specified session_role must be a role that the current login_user is a member of.
+    - Permissions checking for SQL commands is carried out as though
+      the session_role were the one that had logged in originally.
+    type: str
+
+notes:
+- Supports PostgreSQL version 9.4+.
+
+author:
+- Andrew Klychkov (@Andersson007)
+
+extends_documentation_fragment: postgres
+'''
+
+EXAMPLES = r'''
+- name: Vacuum database acme
+  postgresql_vacuum:
+    db: acme
+
+- name: Vacuum and analyze database acme
+  postgresql_vacuum:
+    db: acme
+    analyze: yes
+
+- name: Vacuum full and analyze database foo. Pay attention that the database will be locked
+  postgresql_vacuum:
+    db: foo
+    full: yes
+    analyze: yes
+
+- name: Analyze table mytable in database bar
+  postgresql_vacuum:
+    db: bar
+    table: mytable
+    analyze_only: yes
+
+- name: Vacuum and analyze columns col1, col2, and col3 of mytable in database acme
+  postgresql_vacuum:
+    db: acme
+    table: mytable
+    columns:
+    - col1
+    - col2
+    - col3
+    analyze: yes
+
+- name: Vacuum of mytable with freeze
+  postgresql_vacuum:
+    db: acme
+    table: mytable
+    freeze: yes
+
+- name: Vacuum of mytable with disabled page skipping
+  postgresql_vacuum:
+    db: acme
+    table: mytable
+    disable_page_skipping: yes
+'''
+
+RETURN = r'''
+queries:
+  description: List of executed queries.
+  returned: always
+  type: str
+  sample: [ "VACUUM (ANALYZE) mytable (col1, col2, col3)" ]
+'''
+
+try:
+    from psycopg2.extras import DictCursor
+except ImportError:
+    # psycopg2 is checked by connect_to_db()
+    # from ansible.module_utils.postgres
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.database import pg_quote_identifier
+from ansible.module_utils.postgres import (
+    connect_to_db,
+    exec_sql,
+    postgres_common_argument_spec,
+)
+from ansible.module_utils._text import to_native
+from ansible.module_utils.six import iteritems
+
+
+DISABLE_PAGE_SKIPPING_VER = 96000
+
+
+class Vacuum(object):
+
+    """Implements VACUUM [FULL] and/or ANALYZE PostgreSQL command behavior.
+
+    args:
+        module (AnsibleModule) -- object of AnsibleModule class
+        cursor (cursor) -- cursor objec of psycopg2 library
+
+    attrs:
+        module (AnsibleModule) -- object of AnsibleModule class
+        cursor (cursor) -- cursor objec of psycopg2 library
+        changed (bool) --  something was changed after execution or not
+        executed_queries (list) -- executed queries
+        query_frag (list) -- list for making SQL query
+    """
+
+    def __init__(self, module, cursor):
+        self.module = module
+        self.cursor = cursor
+        self.executed_queries = []
+        self.changed = False
+        self.query_frag = []
+
+    def __check_obj(self):
+        """
+        Check the table or columns that need to be vacuumed/analyzed exist.
```

Comment:
Consider using [information_schema.tables](https://www.postgresql.org/docs/current/infoschema-tables.html) and [information_schema.columns](https://www.postgresql.org/docs/current/infoschema-columns.html) instead. It seems more elegant to me.

Reply:
great, thank you for this hint! changed